excitebike/src/main/java/com/codenjoy/dojo/excitebike/model/items/bike/Bike.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    @Override
+    public void right() {
+        if (!isAlive()) return;
+
+        changeIncline(BikeType.BIKE_INCLINE_RIGHT, BikeType.BIKE_INCLINE_LEFT);
+    }
+
+    private void changeIncline(BikeType toIncline, BikeType inclinedTo) {
+        if (type == BikeType.BIKE) {
+            type = toIncline;
+        } else if (type == inclinedTo) {
+            type = BikeType.BIKE;
+        }
+    }
+
+    public void crush() {
+        type = BikeType.BIKE_FALLEN;
+    }
+
+    public void jump() {
+        //TODO add new BikeType.JUMP for springboard
+        //type = BikeType.BIKE_JUMP;
+    }
+
+    @Override
+    public void act(int... p) {
+        //nothing to do
+    }
+
+    @Override
+    public void tick() {
+        if (isAlive()) {
+            if (direction != null) {
+                int newX = direction.changeX(x);
+                int newY = direction.changeY(y);
+                tryToMove(newX, newY);
+            }
+            interactWithOtherElements(x, y);
+        } else {
+            shift();
+        }
+    }
+
+    private void tryToMove(final int x, final int y) {
+        Optional<Bike> enemyBike = field.getEnemyBike(x, y);
+        enemyBike.ifPresent(this::interactWithOtherBike);
+
+        if (canMoveTo(x, y) && direction != null) {
+            move(x, y);
+            direction = null;
+        }
+    }
+
+    private boolean canMoveTo(final int x, final int y) {
+        return !field.isBorder(x, y);
+    }
+
+    private void interactWithOtherBike(Bike enemyBike) {
+        if (enemyBike.direction == null) {
+            enemyBike.crush();
+            direction = null;
+        } else if (direction != enemyBike.direction) {
+            enemyBike.direction = null;
+            direction = null;
+        }
+    }
+
+    private void interactWithOtherElements(final int x, final int y) {
+        int acceleratorStep = 2;
+        int inhibitorStep = -2;
+
+        if (field.isAccelerator(x, y)) {
+            if (x + acceleratorStep < field.size()) {
+                move(x + acceleratorStep, y);
+            } else {
+                move(field.size() - 1, y);
+            }
+            return;
+        }
+
+        if (field.isInhibitor(x, y)) {
+            if (x + inhibitorStep >= 0) {
+                move(x + inhibitorStep, y);
+            } else {
+                move(0, y);
+            }
+            return;
+        }
+
+        if (field.isObstacle(x, y)) {
+            crush();
+            shift();
+            return;
+        }
+
+        if (field.isUpLineChanger(x, y)) {
+            direction = Direction.UP;
+            tryToMove(x, direction.changeY(y));
+            return;
+        }
+
+        if (field.isDownLineChanger(x, y)) {
+            direction = Direction.DOWN;
+            tryToMove(x, direction.changeY(y));
+            return;
+        }
+
+        if (field.isOffRoad(x, y) && x > 0) {
+            //TODO something like shift()
```

Comment:
what do you mean here?
+ also, leaving todo is ok if there is a task for it only. 
Like basic rule is: we have to implement the todo right now or file a task to do it later.